common/nng/socket.cpp

```diff
@@ -39,14 +39,22 @@ Socket::~Socket()
 
 bool Socket::SyncRequired() { return false; }
 
-int Socket::Connect(const std::string &addr, bool retry)
+int Socket::Connect(const std::string &addr, int retry)
 {
     mapf_assert(!addr.empty());
     int rc;
     do {
         rc = nng_dial(sock->sd_, addr.c_str(), nullptr, 0);
```

Comment:
Not really part of what this patch is trying to fix, but: nng_dial returns immediately on error. Thus it will hammer the bus without any pause. If the current thread happens to have higher priority than the bus itself, the bus will never get any CPU and we are livelocked.

In other words: in case of failure, there should be a small sleep (10ms seems appropriate, 1 jiffy at HZ=100).